File: java/242.valid-anagram.java
/*
 * @lc app=leetcode id=242 lang=java
 *
 * [242] Valid Anagram
 */

// @lc code=start
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] charCounts = new int[26];
        char[] chars = s.toCharArray();

        for(char c: chars) {
            charCounts[c-'a']++;
        }

        chars = t.toCharArray();

        for(char c:chars) {
            charCounts[c-'a']--;
        }

        for(int num: charCounts) {
            if(num != 0) return false;
        }

        return true;
    }
}
// @lc code=end

File: cpp/242.valid-anagram.cpp
/*
 * @lc app=leetcode id=242 lang=cpp
 *
 * [242] Valid Anagram
 */

// @lc code=start
class Solution {
public:
    bool isAnagram(string s, string t) {
        int count[26];

        for(char c : s) {
            count[c - 'a']++;
        }

        for(char c : t) {
            count[c - 'a']--;
        }

        return std::all_of(&count[0], &count[26], [](int c){ return c == 0; });
    }
};
// @lc code=end

